wgpu-hal/src/lib.rs

```diff
@@ -700,20 +827,36 @@ pub trait Queue: WasmNotSendSync {
     ///
     /// Valid usage:
     ///
-    /// - All of the [`CommandBuffer`][cb]s were created from
-    ///   [`CommandEncoder`][ce]s that are associated with this queue.
+    /// - All of the [`CommandBuffer`][cb]s must have been created from
+    ///   [`CommandEncoder`][ce]s that were constructed from the [`Device`][d]
+    ///   associated with this [`Queue`].
+    ///
+    /// - A given [`CommandBuffer`][cb] must only be submitted for execution
+    ///   once, even if prior submissions returned errors.
     ///
-    /// - All of those [`CommandBuffer`][cb]s must remain alive until
-    ///   the submitted commands have finished execution. (Since
-    ///   command buffers must not outlive their encoders, this
-    ///   implies that the encoders must remain alive as well.)
+    /// - All of those [`CommandBuffer`][cb]s must remain alive until the
+    ///   submitted commands have finished execution. (Since command buffers
+    ///   must not outlive their encoders, this implies that the encoders must
+    ///   remain alive as well.)
```

Comment:
nitpick: As the note about lifetime relative to encoders seems _pretty important_, I reject the usage of parentheses, which typically implies tangential information.

```suggestion
    ///   submitted commands have finished execution. Since command buffers
    ///   must not outlive their encoders, this implies that the encoders must
    ///   remain alive as well.
```

I know this is from older documentation, so I'm happy to iterate in my own PR, if you want.

Reply:
The reason I think parentheses were okay is that this is strictly a consequence of rules stated elsewhere, not new information. But omitting the parens is fine with me.